### ISSUE TITLE
Crash when editing a large Markdown file

### ISSUE THREAD

**Issue by winitzki** [lapce/lapce#3581]
## Lapce Version

```
$ lapce --version
Lapce 0.4.2
```
## System information

MacOS Sonoma 14.5, Intel MacBook Pro (16MB RAM)

## Describe the bug

Lapce crashes after adding 2 spaces to a Markdown file.

See attached screen recording.

## Additional information

No plugins are installed. The available Markdown plugin was not installed.
(The behavior is the same when the Markdown plugin was installed.)

As the video shows, the crash happens every time when opening a larger Markdown file (`programming_dhall.md`, 500KB, includes many code blocks and inline LaTeX equations). But the crash does not happen when opening a smaller Markdown file (`README.md`, 16KB).

https://github.com/user-attachments/assets/a77431e4-c484-4377-8134-cd4c8d4ed758

**Comment by dzhou121** [comment]
Is `programming_dhall.md` a public file that you can share?

**Comment by panekj** [comment]
@dzhou121 should be https://github.com/winitzki/scall/blob/master/tutorial/programming_dhall.md

**Comment by winitzki** [comment]
> @dzhou121 should be https://github.com/winitzki/scall/blob/master/tutorial/programming_dhall.md

That's right. Sorry, I was just thinking about adding links to those files. The other file in the video is the README.md from the same project. For that file, there is no crash.
